lua/luasnip/init.lua

```diff
@@ -225,6 +244,17 @@ local function _jump_into_default(snippet)
 	return util.no_region_check_wrap(snippet.jump_into, snippet, 1)
 end
 
+---@class Snippet
+---@field id integer
+---@field name string
+---@field trigger string
+---@field trigger_expand unknown
+---@field description string
+---@field wordTrig boolean
+---@field regTrig boolean
+---@field captures table
+
+---@alias SnipExpand fun(snippet: Snippet, opts: table|nil): Snippet
```

Comment:
Maybe for the future: It would be cool to also have the `opts` to all the functions type-hinted :D
I think luals is capable of eg. suggesting keys if a table is passed, and that would help discovery of options

Reply:
Ah, like you've done in `expand`, nice!

Reply:
Yeah, I did in some places, but as it was too much of a task for a first PR and for my noobness with Luasnip too typehint the expected types!